txircd/ircd.py

```diff
@@ -372,7 +374,111 @@ def reloadModule(self, moduleName):
 		else:
 			deferList.addCallback(lambda result: self.loadModule(moduleName))
 		return deferList
-	
+
+	def verifyConfig(self, config):
+		# Verify the configuration values that are used in the core first.
+
+		# IRCd
+		if not isinstance(config["bind_client"], list):
+			raise ConfigValidationError("bind_client", "value must be a list")
+		for bindDesc in config["bind_client"]:
+			if not isinstance(bindDesc, str):
+				raise ConfigValidationError("bind_client", "every entry must be a string")
+		if not isinstance(config["bind_server"], list):
+			raise ConfigValidationError("bind_server", "value must be a list")
+		for bindDesc in config["bind_server"]:
+			if not isinstance(bindDesc, str):
+				raise ConfigValidationError("bind_server", "every entry must be a string")
+		if not isinstance(config["modules"], list):
+			raise ConfigValidationError("modules", "value must be a list")
+		for module in config["modules"]:
+			if not isinstance(module, str):
+				raise ConfigValidationError("modules", "every entry must be a string")
+		if "links" in config:
+			if not isinstance(config["links"], dict):
+				raise ConfigValidationError("links", "value must be a dictionary")
+			for desc, server in config["links"].iteritems():
+				if not isinstance(desc, str):
+					raise ConfigValidationError("links", "\"{}\" is an invalid server description".format(desc))
+				if not isinstance(server, dict):
+					raise ConfigValidationError("links", "values for \"{}\" must be a dictionary".format(desc))
```

Comment:
We're verifying all the other types; are we not going to validate the keys/values of this dictionary, too? :P  (This includes all the other dictionary checks, too.)